pkg/operation/botanist/component/nodelocaldns/monitoring.go

```diff
@@ -0,0 +1,102 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodelocaldns
+
+import (
+	"strconv"
+	"strings"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/kubeapiserver"
+)
+
+const (
+	monitoringPrometheusJobName = "node-local-dns"
+
+	monitoringMetricBuildInfo                                     = "coredns_build_info"
+	monitoringMetricCacheEntries                                  = "coredns_cache_entries"
+	monitoringMetricCacheHitsTotal                                = "coredns_cache_hits_total"
+	monitoringMetricCacheMissesTotal                              = "coredns_cache_misses_total"
+	monitoringMetricDnsRequestDurationSecondsCount                = "coredns_dns_request_duration_seconds_count"
+	monitoringMetricDnsRequestDurationSecondsBucket               = "coredns_dns_request_duration_seconds_bucket"
+	monitoringMetricDnsResponsesTotal                             = "coredns_dns_responses_total"
+	monitoringMetricForwardRequestsTotal                          = "coredns_forward_requests_total"
+	monitoringMetricForwardResponsesTotal                         = "coredns_forward_responses_total"
+	monitoringMetricKubernetesDnsProgrammingDurationSecondsBucket = "coredns_kubernetes_dns_programming_duration_seconds_bucket"
+	monitoringMetricKubernetesDnsProgrammingDurationSecondsCount  = "coredns_kubernetes_dns_programming_duration_seconds_count"
+	monitoringMetricKubernetesDnsProgrammingDurationSecondsSum    = "coredns_kubernetes_dns_programming_duration_seconds_sum"
+	monitoringMetricProcessMaxFds                                 = "process_max_fds"
+	monitoringMetricProcessOpenFds                                = "process_open_fds"
+)
+
+var (
+	monitoringAllowedMetrics = []string{
+		monitoringMetricBuildInfo,
+		monitoringMetricCacheEntries,
+		monitoringMetricCacheHitsTotal,
+		monitoringMetricCacheMissesTotal,
+		monitoringMetricDnsRequestDurationSecondsCount,
+		monitoringMetricDnsRequestDurationSecondsBucket,
+		monitoringMetricDnsResponsesTotal,
+		monitoringMetricForwardRequestsTotal,
+		monitoringMetricForwardResponsesTotal,
+		monitoringMetricKubernetesDnsProgrammingDurationSecondsBucket,
+		monitoringMetricKubernetesDnsProgrammingDurationSecondsCount,
+		monitoringMetricKubernetesDnsProgrammingDurationSecondsSum,
+		monitoringMetricProcessMaxFds,
+		monitoringMetricProcessOpenFds,
+	}
+
+	// TODO: Replace below hard-coded paths to Prometheus certificates once its deployment has been refactored.
+	monitoringScrapeConfig = `job_name: ` + monitoringPrometheusJobName + `
+ scheme: https
```

Comment:
With this PR, prometheus is failing to start with:

```
level=error ts=2021-12-20T08:49:10.979Z caller=main.go:360 msg="Error loading config (--config.file=/etc/prometheus/config/prometheus.yaml)" err="parsing YAML file /etc/prometheus/config/prometheus.yaml: yaml: line 767: mapping values are not allowed in this context"
```

The issue seems to be that we have additional space:

```suggestion
scheme: https
```